dask_expr/_expr.py

```diff
@@ -461,23 +461,23 @@ def _combine_similar_branches(self, root, remove_ops, skip_ops=None):
             return None
 
         others = self._find_similar_operations(root, ignore=self._parameters)
-        if isinstance(self.frame, Filter) and all(
-            isinstance(op.frame, Filter) for op in others
-        ):
-            # Avoid pushing filters up if all similar ops
-            # are acting on a Filter-based expression anyway
-            return None
```

Comment:
Doing this early doesn't make too much sense, we should only do this for ops that are potentially compatible with self.